src/RobotModelUpdate.cpp

```diff
@@ -11,8 +11,20 @@ RobotModelUpdate::~RobotModelUpdate() = default;
 
 void RobotModelUpdate::init(mc_control::MCGlobalController & controller, const mc_rtc::Configuration & config)
 {
+  auto & ctl = controller.controller();
   mc_rtc::log::info("RobotModelUpdate::init called with configuration:\n{}", config.dump(true, true));
-  robot_ = config("robot", controller.controller().robot().name());
+  if(!ctl.config().has("RobotModelUpdate"))
+  {
+    robot_ = config("robot", controller.controller().robot().name());
+  }
+  else
+  {
+    auto ctlConf = ctl.config()("RobotModelUpdate");
+    ctlConf("robot", robot_);
+  }
+
+  config_.load(config);
+
```

Comment:
We can do better:
1. Store the global plugin configuration in `config_`
2. Merge it with the controller's configuration if it exist
3. Use it

```suggestion
  config_ = config;
  if(auto ctlConfig = ctl.config().find("RobotModelUpdate"))
  {
    config_.load(*ctlConfig);
  }
  robot_ = config_("robot", ctl.robot().name());
```